### DISCUSSION TITLE
[@types/react] Bug related to forms introduced in `^18.2.4`

### DISCUSSION THREAD

**Discussion by sukalov** [DefinitelyTyped/DefinitelyTyped#68137] [Issues with a @types package] [answered]
My app build goes well with `@types/react -v 18.2.3`, but with versions 18.2.4 and later build fails
(up until latest 18.2.47).

```bash
Type error: Type '{ disabled?: boolean | undefined; form?: string | undefined; formAction?: string | ((formData: FormData) => void) | undefined; formEncType?: string | undefined; formMethod?: string | undefined; ... 267 more ...; ref: ForwardedRef<...>; }' is not assignable to type 'SlotProps'.
build:   Types of property 'children' are incompatible.
build:     Type 'React.ReactNode' is not assignable to type 'import("/Users/matvey/Documents/mktour/node_modules/@radix-ui/react-slot/node_modules/@types/react/ts5.0/index").ReactNode'.
build:       Type 'ReactElement<any, string | JSXElementConstructor<any>>' is not assignable to type 'ReactNode'.
build:         Property 'children' is missing in type 'ReactElement<any, string | JSXElementConstructor<any>>' but required in type 'ReactPortal'.
```

showing error at {...props} in line:
```bash
build: > 43 |                   <Comp className={cn(buttonVariants({ variant, size, className }))} ref={ref} {...props} />
```

- typescript version 5.0.4 (later typescripts produce different bugs even with 18.2.3)

[a comment by an automated account is not shown]

**Comment by Semigradsky** [comment]
You have several `@types/react` versions in node_modules. Remain one

**Reply by eps1lon** [reply, marked as answer]
For more info, check out https://solverfox.dev/writing/duplicate-types-react

**Reply by sukalov** [reply]
Thanks for answering! Are you referring to `@types/react-dom`? I've uninstalled it and then reinstall all the `node modules` to avoid accidental duplicates, but still, after I try updating `@types/react` it throws same error

**Reply by eps1lon** [reply]
>Thanks for answering! Are you referring to @types/react-dom?

And `@types/react`

**Reply by sukalov** [reply]
Thanks a lot!! 
I just added an override to latest, and it worked great!
